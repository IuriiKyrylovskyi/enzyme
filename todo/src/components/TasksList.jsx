import React from "react";
import { PropTypes } from "prop-types";
import Task from "./Task";

const TasksList = ({ tasks, handleStatusChange, handleTaskDelete }) => {
  // const sortedTasks = tasks.slice().sort((a, b) => a.done - b.done);

  return (
    <ul className="list">
      {tasks.length > 0 &&
        tasks.map((task) => {
          const { id, done, text } = task;

          return (
            <Task
              key={id}
              id={id}
              done={done}
              text={text}
              handleStatusChange={handleStatusChange}
              handleTaskDelete={handleTaskDelete}
              //
            />
          );
        })}
    </ul>
  );
};

TasksList.propTypes = {
  tasks: PropTypes.arrayOf(PropTypes.shape()),
  handleStatusChange: PropTypes.func.isRequired,
  handleTaskDelete: PropTypes.func.isRequired,
};

export default TasksList;
