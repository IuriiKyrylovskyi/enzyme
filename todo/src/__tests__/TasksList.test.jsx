import React from "react";
import { shallow } from "enzyme";
import TasksList from "../components/TasksList";

describe("<TasksList />", () => {
  it("should display tasks list sorted by 'done' status", () => {
    const props = {
      handleStatusChange: jest.fn(),
      handleTaskDelete: jest.fn(),
      tasks: [
        {
          id: "id-1",
          done: true,
          text: "Task 1",
        },
        {
          id: "id-2",
          done: false,
          text: "Task 2",
        },
        {
          id: "id-3",
          done: true,
          text: "Task 3",
        },
      ],
    };
    const wrappedComponent = shallow(<TasksList {...props} />);

    expect(wrappedComponent).toMatchSnapshot();
  });
});
