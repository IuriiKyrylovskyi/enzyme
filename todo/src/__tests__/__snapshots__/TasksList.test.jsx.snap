// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksList /> should display tasks list sorted by 'done' status 1`] = `
<ul
  className="list"
>
  <Task
    done={true}
    handleStatusChange={[MockFunction]}
    handleTaskDelete={[MockFunction]}
    id="id-1"
    key="id-1"
    text="Task 1"
  />
  <Task
    done={false}
    handleStatusChange={[MockFunction]}
    handleTaskDelete={[MockFunction]}
    id="id-2"
    key="id-2"
    text="Task 2"
  />
  <Task
    done={true}
    handleStatusChange={[MockFunction]}
    handleTaskDelete={[MockFunction]}
    id="id-3"
    key="id-3"
    text="Task 3"
  />
</ul>
`;
